src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -80,6 +81,10 @@ public void lock(UUID datasetId, String flightId) {
 
         // if no rows were updated, then throw an exception
         if (numRowsUpdated == 0) {
+            // try to retrieve the dataset in case we should throw a NotFound exception
+            retrieveSummaryById(datasetId);
```

Comment:
What is this here for? Is some logic missing?

Reply:
This method is to check if the dataset exists. If it does not, then the method throws a DatasetNotFound exception. I don't need the result (dataset summary) here, just the existence check, so I ignore the return value.

Reply:
Ah, got it. Maybe add a comment?

Reply:
Done.